snapcraft/internal/sources.py

```diff
@@ -325,17 +326,26 @@ def provision(self, dst, clean_target=True, keep_zip=False):
 class Local(Base):
 
     def pull(self):
-        if os.path.islink(self.source_dir):
+        if os.path.islink(self.source_dir) or os.path.isfile(self.source_dir):
             os.remove(self.source_dir)
-        elif (os.path.isdir(self.source_dir) and
-              not os.listdir(self.source_dir)):
-            os.rmdir(self.source_dir)
-        elif os.path.exists(self.source_dir):
-            raise EnvironmentError('Cannot pull to target {!r}'.format(
-                self.source_dir))
+        elif os.path.isdir(self.source_dir):
+            shutil.rmtree(self.source_dir)
```

Comment:
Take particular note here: previously the code was very careful not to overwrite an existing `parts/<name>/src` directory or file. That logic is essentially impossible now that we're not symlinking, so I decided to assert ownership over `parts/<name>/` and just blow it away if it already exists. See any downside to that? Users shouldn't be putting things there...